Tests/UnitTests/Tests/HttpTests.cpp

```diff
@@ -454,69 +454,52 @@ DEFINE_TEST_CLASS(HttpTests)
         HCCleanup();
     }
 
-    // Temporarily enabled for testing WinHTTP flag changes
-    DEFINE_TEST_CASE(TestGdkHttpRequest)
+    DEFINE_TEST_CASE(TestHttpProtocol)
     {
-        DEFINE_TEST_CASE_PROPERTIES(TestGdkHttpRequest);
+        DEFINE_TEST_CASE_PROPERTIES(TestHttpProtocol);
         
-        // GDK PC integration test to verify HTTP (non-HTTPS) requests work correctly
+        // Test to verify HTTP (non-HTTPS) requests work correctly
         // 
-        // Background: Previously, GDK builds used WINHTTP_FLAG_SECURE_DEFAULTS for all requests,
-        // which could block HTTP requests. The fix ensures HTTP URLs use WINHTTP_FLAG_ASYNC
-        // while HTTPS URLs continue to use WINHTTP_FLAG_SECURE_DEFAULTS.
-        //
-        // This test verifies that HTTP requests can proceed to the network layer without
-        // being blocked by WinHTTP flag configuration issues.
+        // Background: This test was added to ensure HTTP protocol support remains functional
+        // after changes to provider flag configuration that could potentially affect HTTP vs HTTPS handling.
         
         VERIFY_ARE_EQUAL(S_OK, HCInitialize(nullptr));
 
         HCCallHandle call = nullptr;
         VERIFY_ARE_EQUAL(S_OK, HCHttpCallCreate(&call));
 
-        // Use an HTTP URL (not HTTPS) to test the WinHTTP flags behavior
-        // Using a local/private IP that won't resolve to test flag configuration without external dependency
-        VERIFY_ARE_EQUAL(S_OK, HCHttpCallRequestSetUrl(call, "GET", "http://192.168.1.255/test"));
+        // Use HTTP protocol to verify it works correctly
+        VERIFY_ARE_EQUAL(S_OK, HCHttpCallRequestSetUrl(call, "GET", "http://example.com"));
+        VERIFY_ARE_EQUAL(S_OK, HCHttpCallRequestSetRetryAllowed(call, false));
         
-        // Set a short timeout since we expect this to fail with network error, not WinHTTP flag error
-        VERIFY_ARE_EQUAL(S_OK, HCHttpCallRequestSetTimeout(call, 1));
+        // Create a mock response for the HTTP call
```

Comment:
I'm not familiar with the HC Mocking stuff. if you use a mock is it still actually testing that WinHTTP is allowing HTTP protocol usage? I think part of my example.com feedback was that you could make a real http request not a mock one

Reply:
unit tests are all against mocks and don't make real calls so I modified the GDK sample to verify its working e2e